### DISCUSSION TITLE
Book 3: Wouldn't we want to skip pdf for constant medium's phase functions?

### DISCUSSION THREAD

**Discussion by StrangeVoyager** [RayTracing/raytracing.github.io#1290] [Debugging]
I am converting my RTWeekend from the original Kindle books, so I may have messed something up. I tried making the SpherePDF for the Isotropic phase function, and it dramatically darkens the medium if I use the mixture PDF of light and srec.pdf_ptr. However, if I set srec to skip_pdf in IsotropicPhase, then it looks correct. I experienced this also with the Henyey-Greenstein phase function (implemented from Jensen's book) and I am assuming I will with the Schlick phase function (but I wanted to check in before I proceed). If we are going to have scattering_pdf (in the phase function) and value (in the PDF) use the same formula, for the same direction vector, why wouldn't we just mark it as specular/skip_pdf?

**Comment by hollasch** [comment]
Converted to issue #1291 for investigation.
